src/GAP.jl

```diff
@@ -419,6 +420,35 @@ function prompt()
     reset_GAP_ERROR_OUTPUT()
 end
 
+"""
+    GAP.randseed!([seed::Integer])
+
+Reseed GAP's global RNG with `seed`.
+
+The given `seed` must be a non-negative integer.
+When `seed` is not specified, a random seed is generated from Julia's global RNG.
+
+For a fixed seed, the stream of generated numbers is allowed to change between
+different versions of GAP.
+"""
+function randseed!(seed::Union{Integer,Nothing}=nothing)
+    if seed === nothing
+        seed = rand(UInt128)
+    end
+    # Although GAP's `Reset` method can accept directly integers (also negative
+    # ones), we harmonize here the behavior with what Julia does, i.e. for two
+    # integers with the same value, even of different types, the initialization
+    # is the same (e.g. with `1` or `0x1`); so we re-use Julia's
+    # `Random.make_seed` internal function (which errors on negative argument)
+    # to normalize this, and create a string out of it.
+    str = String(reinterpret(UInt8, Random.make_seed(seed)))
+    Globals.Reset(Globals.GlobalMersenneTwister, str)
+    # when GlobalRandomSource is reset, the seed is taken modulo 2^28, so we just
+    # pass an already reduced seed here
+    Globals.Reset(Globals.GlobalRandomSource, seed % Int % 2^28)
+    nothing
```

Comment:
This looks fine now to me, thanks.

That said... Is there a way in Julia to capture the current RNG state for e.g. serialization, and to possibly restore it later? If so, then returning that here might be reasonable. (But I am *not* requesting that you do this here; I am just musing about this; if at all, this could come in a future PR, or never)

Reply:
In Julia, the "idiom" (quotes, as I'm not sure many people know it!) for capturing the RNG state is simply to call `copy` on the RNG object, and the state can be restored later with `copy!`:
```julia

julia> rng = MersenneTwister(); rand(rng); rng2 = copy(rng); rng == rng2
true

julia> rand(rng) == rand(rng2)
true

julia> rand(rng); copy!(rng, rng2); rng == rng2
true
```
Here, there is a complication, because two global RNGs are involved. With one RNG, my take would be to have a Julia RNG wrapper (subtype of `AbstractRNG`) around GAP's `IsMersenneTwister` and implement `copy` / `copy!`. Then, like `Random.seed!`, `GAP.randseed!` could simply return an RNG object which wraps GAP's `GlobalMersenneTwister`. But given that two RNGs are involved here, it seems simpler to return `nothing`, but we could very well wrap both RNG types and return a tuple containing both RNGs.